p2p/simulations/http.go

```diff
@@ -380,6 +381,13 @@ func (s *Server) ResetNetwork(w http.ResponseWriter, req *http.Request) {
 
 // StreamNetworkEvents streams network events as a server-sent-events stream
 func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
+	var throttle <-chan time.Time
+	//When running a sim, the frontend may "choke to death" depending on params (node count)
+	//due to overwhelming event rate. It is therefore important to be able to throttle
+	//the rate at which the events are emitted, in order for the frontend to have time
+	//to terminate rendering before the next event arrives.
+	throttling := false
```

Comment:
I don't really agree with this, can the events not be buffered and dropped client side? It seems that is where the issue is here.

Reply:
Well excellent objection. My biggest argument is that it is much easier to do it here in go than in the frontend in javascript. Furthermore, the javascript client needs to subscribe to the event stream directly with the HTTP Server-side Eventstream API, thus it gets flooded with all the requests and would need to handle them. Javascript being singlethreaded, apart from more complex in the implementation, such an approach doesn't relieve the frontend from the load, which is the main reason to do this throttling in the first place. An effective way to do this client side would probably be building some kind of queue, which may imply additional processes and more moving pieces (I indeed started a first attempt with this in mind, but then ditched it).

In short - this is way easier, effective and efficient to do here in the backend, Also, it is optional and disabled by default, only if provided in the query string it will be activated. 

BTW the events are not being dropped but throttled.